EmbeddedScripts.CSharp.Tests/ScriptCodeRunnerTests.cs

```diff
@@ -0,0 +1,80 @@
+using Xunit;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Scripting;
+using HelperObjects;
+using System;
+using EmbeddedScripts.CSharp.Scripting;
+using EmbeddedScripts.Shared;
+
+namespace EmbeddedScripts.CSharp.Tests
+{
+    public class ScriptCodeRunnerTests
+    {
+        [Fact]
+        public async Task CorrectScriptCode_ExecutionSucceed()
+        {
+            var code = "var a = 1; var b = 2; var c = a + b;";
+
+            var runner = new ScriptCodeRunner(code, new Container());
+
+            await runner.RunAsync();
+        }
+
+        [Fact]
+        public async Task RunWithGlobalVariables_Succeed()
+        {
+            var container = new Container();
+
+            var t = new HelperObject();
+
+            container.Register(t, "t");
+
+            var code = "t.x++;";
+
+            var runner = new ScriptCodeRunner(code, container);
+            await runner.RunAsync();
+
+            Assert.Equal(1, t.x);
+        }
+
+        [Fact]
+        public async Task RunWithFuncFromContainer_Succeed()
+        {
+            var container = new Container();
+
+            int x = 0;
+            container.Register<Action>(() => { x++; }, "t");
+
+            var code = "t();";
+
+            var runner = new ScriptCodeRunner(code, container);
+            await runner.RunAsync();
+
+            Assert.Equal(1, x);
+        }
+
+
+        [Fact(Skip = "test is skipped until security question is resolved")]
+
+        public async Task RunWithNotAllowedFunc_Fail()
+        {
+            var container = new Container();
+            
+            var code = "var w = new System.IO.StreamWriter(\"a.txt\"); w.WriteLine(\"Hello\"); w.Close();";
+
+            var runner = new ScriptCodeRunner(code, container);
+
+            await Assert.ThrowsAsync<CompilationErrorException>(() => runner.RunAsync());
+        }
+
+        [Fact]
+        public async Task InvalidScriptCode_ThrowsAnException()
```

Comment:
You have covered only compilation failures, but what about runtime failures? For example, when you cast to a wrong type or devide by 0 in the usercode?